src/accumulator/stump.rs

```diff
@@ -0,0 +1,139 @@
+use super::types;
+
+#[derive(Debug, Clone)]
+pub struct Stump {
+  leafs: u64,
+  roots: Vec<bitcoin_hashes::sha256::Hash>
+}
+
+impl Stump {
+  /// Creates an empty Stump
+  pub fn new() -> Self {
+    Stump {
+      leafs: 0,
+      roots: Vec::new()
+    }
+  }
+  /// Modify is the external API to change the accumulator state. Since order
+  /// matters, you can only modify, providing a list of utxos to be added, 
+  /// and txos (@TODO) to be removed, along with it's proof. Either may be
+  /// empty.
+  ///# Example
+  /// ```
+  ///   use rustreexo::accumulator::stump::Stump;
+  ///   let mut s = Stump::new();
+  ///   let utxos = vec![];
+  ///   let stxos = vec![];
+  ///   s.modify(&utxos, &stxos);
+  /// ```
+  pub fn modify(&mut self, utxos: &Vec<bitcoin_hashes::sha256::Hash>, _stxos: &Vec<bitcoin_hashes::sha256::Hash>) {
```

Comment:
Could we change to `self` instead of `&mut self` for all these functions? Stump doesn't need to be mutated.

Reply:
I see what you mean. It's actually a good point.  
If modify is a pure function, then stump doesn't need to be mutable, every time you update a stump you create a new one and assign to a new variable. You may even shadow out the older stump, so you can use the same name after modifying.  
I'll experiment with that idea a little.

Reply:
Do you wanna do that in this PR or in a follow up PR?